Community/Tdarr_Plugin_MC93_Migz5ConvertAudio.js

```diff
@@ -72,93 +88,113 @@ const plugin = (file, librarySettings, inputs, otherArguments) => {
+            languageAudioStreams.hasChannel6 = true;
+            has6Channel = true;
+            break;
+          case 2:
+          default:
+            languageAudioStreams.channel2 = { index: i, codecName: file.ffProbeData.streams[i].codec_name };
+            languageAudioStreams.hasChannel2 = true;
+            has2Channel = true;
+            break;
         }
       }
     } catch (err) {
       // Error
     }
   }
 
-  // Go through each stream in the file.
-  for (let i = 0; i < file.ffProbeData.streams.length; i++) {
-    // Check if stream is audio.
-    if (file.ffProbeData.streams[i].codec_type.toLowerCase() === 'audio') {
-      // Catch error here incase user left inputs.downmix empty.
-      try {
-        // Check if inputs.downmix is set to true.
-        if (inputs.downmix.toLowerCase() === 'true') {
-          // Check if file has 8 channel audio but no 6 channel, if so then create extra downmix from the 8 channel.
-          if (
-            has8Channel === true
-            && has6Channel === false
-            && file.ffProbeData.streams[i].channels === 8
-          ) {
+  if ((downmix && !downmixLanguageAware) || aacStereo)
+    for (let i = 0; i < file.ffProbeData.streams.length; i++)
+      if (file.ffProbeData.streams[i].codec_type.toLowerCase() === 'audio') {
+        if (downmix && !downmixLanguageAware) {
+          if (file.ffProbeData.streams[i].channels === 8 && !has6Channel) {
             ffmpegCommandInsert += `-map 0:${i} -c:a:${audioIdx} ac3 -ac 6 -metadata:s:a:${audioIdx} title="5.1" `;
             response.infoLog += '☒Audio track is 8 channel, no 6 channel exists. Creating 6 channel from 8 channel. \n';
             convert = true;
           }
-          // Check if file has 6 channel audio but no 2 channel, if so then create extra downmix from the 6 channel.
-          if (
-            has6Channel === true
-            && has2Channel === false
-            && file.ffProbeData.streams[i].channels === 6
-          ) {
+          if (file.ffProbeData.streams[i].channels === 6 && !has2Channel) {
             ffmpegCommandInsert += `-map 0:${i} -c:a:${audioIdx} aac -ac 2 -metadata:s:a:${audioIdx} title="2.0" `;
             response.infoLog += '☒Audio track is 6 channel, no 2 channel exists. Creating 2 channel from 6 channel. \n';
             convert = true;
           }
         }
-      } catch (err) {
-        // Error
+
+        if (
+          aacStereo
+          && file.ffProbeData.streams[i].channels === 2
+          && file.ffProbeData.streams[i].codec_name !== 'aac'
+        ) {
+          ffmpegCommandInsert += `-c:a:${audioIdx} aac `;
+          response.infoLog += '☒Audio track is 2 channel but is not AAC. Converting. \n';
+          convert = true;
+        }
+
+        ++audioIdx;
       }
 
-      // Catch error here incase user left inputs.downmix empty.
+  if (downmix && downmixLanguageAware)
```

Comment:
if statement is missing its brackets

Reply:
The only statement in the body of this if is a for loop. So it's ok to not have the brackets.
But I suppose that your request is for readability purposes so I added them anyway.